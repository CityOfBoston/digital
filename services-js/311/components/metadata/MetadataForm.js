// @flow

import React from 'react';
import { css } from 'glamor';

import type { ServiceMetadata } from '../../store/modules/services';

export type Props = {|
  attributes: $PropertyType<$NonMaybeType<ServiceMetadata>, 'attributes'>,
|}

const STYLE = {
  textTextarea: css({
    width: '100%',
    height: 180,
  }),
};

function renderInformationalAttribute(attribute) {
  return (
    <p key={attribute.code}>{attribute.description}</p>
  );
}

function renderTextAttribute(attribute) {
  return (
    <label key={attribute.code}>
      <p>{attribute.description}</p>
      <textarea className={STYLE.textTextarea} />
    </label>
  );
}

function renderPicklistAttribute(attribute) {
  return (
    <label key={attribute.code}>
      <p>{attribute.description}</p>
      <select name={attribute.code}>
        {(attribute.values || []).map(({ key, value }) => <option value={key} key={key}>{value}</option>)}
      </select>
    </label>
  );
}

function renderAttribute(attribute) {
  switch (attribute.type) {
    case 'informational':
      return renderInformationalAttribute(attribute);
    case 'text':
      return renderTextAttribute(attribute);
    case 'picklist':
      return renderPicklistAttribute(attribute);
    default:
      return null;
  }
}

export default function MetadataForm({ attributes }: Props) {
  return (
    <div>
      {(attributes || []).map(renderAttribute)}
    </div>
  );
}
