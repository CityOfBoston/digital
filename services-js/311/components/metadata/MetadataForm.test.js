// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import MetadataForm from './MetadataForm';

const ATTRIBUTES = [
  {
    required: false,
    type: 'text',
    order: null,
    description: 'Please provide any other relevant information:',
    code: 'ST-CMTS',
    values: null,
  }, {
    required: false,
    type: 'informational',
    order: 1,
    description: '**All needle pickup cases should be followed up with a phone call to one of the below agencies.**',
    code: 'INFO-NEDRMV1',
    values: null,
  }, {
    required: false,
    type: 'informational',
    order: 2,
    description: 'Mobile Sharps Collection Team: 617-534-7123 (Monday-Friday 8am-6pm)',
    code: 'INFO-NEDRMV2',
    values: null,
  }, {
    required: false,
    type: 'informational',
    order: 3,
    description: 'Boston EMS: 617-343-1400 (All hours outside of Monday-Friday 8am-6pm)',
    code: 'INFO-NEDRMV3',
    values: null,
  }, {
    required: true,
    type: 'picklist',
    order: 4,
    description: 'How many needles are at the location?',
    code: 'SR-NEDRMV1',
    values: [{
      key: 'One',
      value: 'One',
    }, {
      key: 'Two',
      value: 'Two',
    }, {
      key: 'Three',
      value: 'Three',
    }, {
      key: 'More than Three',
      value: 'More than Three',
    }],
  }, {
    required: true,
    type: 'picklist',
    order: 5,
    description: 'Property location type',
    code: 'ST-PROPLOC',
    values: [{
      key: 'Private',
      value: 'Private',
    }, {
      key: 'Public',
      value: 'Public',
    }, {
      key: 'Unknown',
      value: 'Unknown',
    }],
  },
];

test('snapshot', async () => {
  const component = renderer.create(<MetadataForm attributes={ATTRIBUTES} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
