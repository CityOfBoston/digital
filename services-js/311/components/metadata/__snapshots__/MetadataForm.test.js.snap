exports[`test snapshot 1`] = `
<div>
  <label>
    <p>
      Please provide any other relevant information:
    </p>
    <textarea
      className={
        Object {
          "data-css-1bm19ha": "",
        }
      } />
  </label>
  <p>
    **All needle pickup cases should be followed up with a phone call to one of the below agencies.**
  </p>
  <p>
    Mobile Sharps Collection Team: 617-534-7123 (Monday-Friday 8am-6pm)
  </p>
  <p>
    Boston EMS: 617-343-1400 (All hours outside of Monday-Friday 8am-6pm)
  </p>
  <label>
    <p>
      How many needles are at the location?
    </p>
    <select
      name="SR-NEDRMV1">
      <option
        value="One">
        One
      </option>
      <option
        value="Two">
        Two
      </option>
      <option
        value="Three">
        Three
      </option>
      <option
        value="More than Three">
        More than Three
      </option>
    </select>
  </label>
  <label>
    <p>
      Property location type
    </p>
    <select
      name="ST-PROPLOC">
      <option
        value="Private">
        Private
      </option>
      <option
        value="Public">
        Public
      </option>
      <option
        value="Unknown">
        Unknown
      </option>
    </select>
  </label>
</div>
`;
